File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    float dollars = 0;
    int coins = 0;

    do
    {
        dollars = get_float("Change owned: ");      //Taking amount in dollars (float)
    }
    while (dollars < 0);

    dollars = dollars * 100;
    int cents = round(dollars);   //round used here to get correct values
    //Conditions to calculate exact coins
    do
    {
        if (cents >= 25)        //if cents greater than 25 detuct them first
        {
            cents = cents - 25;
            coins++;
        }
        else if (cents >= 10)    //if cents greater than 10 detuct them first
        {
            coins++;
            cents = cents - 10;
        }
        else if (cents >= 5)      //if cents greater than 5 detuct them first
        {
            coins++;
            cents = cents - 5;
        }
        else if (cents >= 1)      //if cents greater than 25 detuct them first
        {
            coins++;
            cents = cents - 1;
        }
        else                     //if on reamining cash left then break
        {
            break;
        }
    }
    while (cents != 0);
    printf("%i",    coins);    //Printing coins
    printf("\n");

}

File: pset6/cash/cash.py
from cs50 import get_float


def main():

    dollars = 0.0
    coins = 0
    cents = 0
    dollars = get_float("Change owned: ")      # Taking amount in dollars (float)
    while (dollars < 0):
        dollars = get_float("Change owned: ")       # Taking amount in dollars (float)
    dollars = dollars * 100
    cents = round(dollars)   # round used here to get correct values
    # Conditions to calculate exact coins
    while (cents != 0):
        if (cents >= 25):        # if cents greater than 25 detuct them first
            cents = cents - 25
            coins += 1
        elif (cents >= 10):    # if cents greater than 10 detuct them first
            coins += 1
            cents = cents - 10
        elif (cents >= 5):      # if cents greater than 5 detuct them first
            coins += 1
            cents = cents - 5
        elif (cents >= 1):      # if cents greater than 25 detuct them first
            coins += 1
            cents = cents - 1
        else:                     # if on reamining cash left then break
            break
    print(coins)    # Printing coins


if __name__ == "__main__":
    main()